servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -86,75 +86,64 @@ class Http4sServlet[F[_]](
   private def onParseFailure(
       parseFailure: ParseFailure,
       servletResponse: HttpServletResponse,
-      bodyWriter: BodyWriter[F]): F[Unit] = {
-    val response =
-      F.pure(
-        Response[F](Status.BadRequest)
-          .withEntity(parseFailure.sanitized))
-    renderResponse(response, servletResponse, bodyWriter)
+      bodyWriter: BodyWriter[F],
+  ): F[Unit] = {
+    val response = Response[F](Status.BadRequest).withEntity(parseFailure.sanitized)
+    renderResponse(response, servletResponse, bodyWriter, F.async(_ => ()))
   }
 
   private def handleRequest(
       ctx: AsyncContext,
       request: Request[F],
       bodyWriter: BodyWriter[F]): F[Unit] = {
-    ctx.addListener(new AsyncTimeoutHandler(request, bodyWriter))
-    // Note: We're catching silly user errors in the lift => flatten.
+    val timeout = F.async[Unit] { cb =>
+      ctx.addListener(new AsyncTimeoutHandler(cb))
+    }
     val response = Async.shift(executionContext) *>
       optionTSync
         .suspend(serviceFn(request))
         .getOrElse(Response.notFound)
         .recoverWith(serviceErrorHandler(request))
-
     val servletResponse = ctx.getResponse.asInstanceOf[HttpServletResponse]
-    renderResponse(response, servletResponse, bodyWriter)
+    F.race(timeout, response).flatMap {
+      case Left(()) =>
+        // The F.never is so we don't interrupt the rendering of the timeout response
```

Comment:
Which `F.never`?

Reply:
Ok, but from the comment I thought is was already being used here, and I don't see it anywhere :grinning:

Reply:
We should leave it in IMHO. It _will_ be never very soon.